[Altair] Update to latest spec v1.1.0-alpha.6

michaelsproul opened the pull request:
## Proposed Changes

* Update our Altair consensus code to v1.1.0-alpha.6
* Add new transition tests, new epoch processing tests, and read metadata file that wasn't read as part of genesis validity tests.
* Update config handling to use the presets and configs defined by the EF spec.

## Additional Info

* Other alpha.6 changes will need to occur in their relevant PRs, e.g. the changes to gossip verification should go in #2321.

michaelsproul committed:
Update consensus code to v1.1.0-alpha.6

michaelsproul committed:
Ripple spec updates

michaelsproul committed:
Implement preset <> config distinction

michaelsproul committed:
Return ConfigAndPreset from VC API

michaelsproul committed:
Don't panic if there aren't any aggregators

Rustfmt also returned with a vengeance to rearrange this block.

michaelsproul committed:
Check state root in transition tests

michaelsproul committed:
Fix clippy

michaelsproul committed:
Add compatibility shim for /config/spec

michaelsproul commented:
Ready for review.

I'm going to experiment now with trying to make the `/config/spec` endpoint backwards compatible with validator clients running Lighthouse v1.4.x and earlier. I initially thought that breaking backwards compatibility would be necessary, but I think we can fix it. For background see `#apis` on Eth R&D: https://discord.com/channels/595666850260713488/710831784991916072/849550587090436126

michaelsproul committed:
Fix Pyrmont & Prater configs. Delete Toledo.

michaelsproul commented:
Ok. This is really ready for review now.

I'm satisfied with the compatibility shim. It only adds a few fields manually to `extra_fields`, and can be switched off via `--http-disable-legacy-spec` on the BN (I haven't added an equivalent option to the VC).

In the process of testing I also realised I'd corrupted the testnet configs. They should be fixed now (I've checked that the specs agree between Lighthouse VC v1.3.0 and this branch, for Pyrmont and Prater).

paulhauner commented on common/eth2_network_config/src/lib.rs:
```diff
@@ -260,11 +248,7 @@ mod tests {
-                || net.name == "pyrmont"
-                || net.name == "prater"
-            {
+            if net.name == "mainnet" || net.name == "pyrmont" || net.name == "prater" {
```
Perhaps we can drop this `if` statement? IIRC it's a relic from when we had v0.12 "legacy" specs hanging around.

paulhauner commented on testing/ef_tests/tests/tests.rs:
```diff
@@ -33,14 +34,17 @@ fn config_test<E: EthSpec + TypeName>() {
 }
 
 #[test]
+#[should_panic]
```
Just a heads-up that there's a `should_panic` are here and on L43.

These will *probably* get refactored once the configs are released, but it seemed like a risk of "deleted the block-comment tags and tests give green tick so they must have passed".

Feel free to ignore.

paulhauner reviewed (approved):
Looks great! I couldn't find anything that needs changing :rocket: I left two small notes, but I'd be happy to merge without them being addressed.

michaelsproul committed:
Minor cleanups

michaelsproul commented:
Thanks for review! I've addressed all comments.

I'll merge this wholesale into `altair` to simplify the diff on the upstream PRs (they'll recognise the existing commits)
